### ISSUE TITLE
service integration (or service onboarding) backlog

### ISSUE THREAD

**Issue by pcrespov** [ITISFoundation/osparc-simcore#2409]
This issue collects a feature "wish list" to help improving the integration of user services in osparc:
![image.png](https://images.zenhubusercontent.com/5caef818ecad11531cc41364/01144ca0-cdab-4e79-9cc7-30d299bd0dbe)
![image.png](https://images.zenhubusercontent.com/5caef818ecad11531cc41364/1c1bbb24-4b6c-42d6-8e6f-f17e179bed7c)
 ![image.png](https://images.zenhubusercontent.com/5caef818ecad11531cc41364/079e43b8-d5c0-478a-9cf0-26fcdbc09981)
### User service codebase

Every service shall have its own github-like repository with 
- the code or at least some reference to it (e.g. Dockerfile that pulls the code, an executable, etc)
- defines ownership (e.g. collaborator, contact, owner, ...)
- issue system in-place for bugs/new features 
- doc (e.g. github pages)
- released versions (might be or not released to osparc)
- MINIMAL osparc publication config  (e.g. all in a ``.osparc`` folder)
- easy testing:   integration library to simplify testing OR. github actions-like to tests 
- easy build: analogous as testing
- easy submission to publication: submit a link github repo + contract ?

### User sandbox

A secure place with limited resources in osparc where the user can test his/her service before open publication

### UI
- Display user-level info of the server in the front-end
      - Ownership: who is behind and contact
      - Help resources: link to forum,  doc, tutorials (e.g. a [tutorial](http://docs.osparc.io/#/docs/tutorials/python_runner) from the python runner)
      - Support issues: link to issue track system

### osparc publication config/contract
- public (for the end-user), semantic (for osparc to classify the upgrade) and integration (for osparc to determine compatibility) versions
- automatic validation workflow
- Upgrade policy contract (see ideas in #2244)

### simplify cookie-cutter
 - service-integration library

### versioning
a single version number does not suffice. Instead we should have a set of versions that describes "what is inside the container"
- service version (following semantic versioning): for the published service
- service integration version: sidecar
- executable name: the public name of the wrapped program (e.g. matlab)
- executable version: the version of the program (e.g. matlab 2020b)
- further libraries version dump (e.g. requirements.txt, etc)

### Identify a service
- can send a link to anybody to refer a published service in a given version (similar to studies)
- 

### Validation tests for publication
Automatic validation test based on service-integration library
- create git-actions to auto CI in github repos
- unit tests
    - service key < 30 characters ( docker limitation for container names is 63 ... which adds uuid)
    - unique title (it does not show twice)
    - ... 

### Image labeling
  - http://label-schema.org/rc1/
  - https://opencontainers.org/

**Comment by GitHK** [comment]
Triage by @KZzizzle:

- jupyter-math 
- jupyter-smash 
- jupyter-neuron 
- jupyter-math-voila 
- jupyter-smash-voila 
- sim4life 
- iseg-app 
- iseg-web 
- jupyter-ml-tensorflow 
- jupyter-ml-pytorch 
- jupyter-chaste 
- jupyter-fenics 
- jupyter-pysonic 
- electrode-selector 
- sim_gui-app 
- sim_gui-web 
- 3d-viewer-gpu 
- 3d-viewer 
- dsistudio-app 
- dsistudio-web 
- raw-graphs 
- bio-formats-app 
- bio-formats-web 
- openmicroscopy-web 
- openmicroscopy-app 
- semcad-app 
- semcad-web 
- bios-health 
- bios-health-vns-calibrator 
- meaplacement-web 
- meaplacement-app 


deprecated (no one should use these anymore - possibly remove from registry):
- jupyter-octave-python-math-voila 
- jupyter-base-notebook 
- jupyter-scipy-notebook 
- raw-graphs-table 
- jupyter-octave-python-math-test 
- jupyter-octave-python-math 
- jupyter-r-notebook 

not accessible in frontend (I think these are not in the catalog for most users):
- dy-static-file-server--sidecar 
- s4l-core 
- sim4life-dy 
- httpbin 
- btl-pc 
- code-server 
- jupyter-ml-base 
- dy-static-file-server--sidecar-compose-spec \
- httpbin--sidecar 
- s4l-core-dy 
- dy-static-file-server 
- jupyter-dask 
- httpbin--sidecar-compose 

too old (probably more trouble than they're worth):
- mattward-viewer 
- bornstein-viewer
- cc-0d-viewer 
- cc-2d-viewer 
- cc-1d-viewer 
- kember-viewer 
- mapcore-widget

**Comment by GitHK** [comment]
## Status Report after Rudolph

-  **jupyter-math** version 2 is has been published to master

**Comment by pcrespov** [comment]
@sanderegg @GitHK @mguidon @odeimaiz @KZzizzle  I disabled dependabot in ``osparc-services`` to avoid having unnecessary PRs that for the moment nobody is handling. I suggest that when the new integration setup is ready, the new versions of the services there:
 - are moved to separate repositories (and an owner assigned)
 - if service is front-end, we could use functional nodes (i.e. does not require repo/container)
 - owner can decide to maintain dependencies/vulnerabilitiess
 - we can consume that info and show it in the released version as well

**Comment by pcrespov** [comment]
![image](https://user-images.githubusercontent.com/32402063/199943187-ae8136b9-91df-454b-86c4-21770b4afbe3.png)
From https://www.ibm.com/docs/en/acvfc?topic=manager-product-lifecycle

**Comment by GitHK** [comment]
@pcrespov @mrnicegyu11 @odeimaiz @elisabettai is this issue still relevant or should we close it?
